### ISSUE TITLE
Credit table not rendering

### ISSUE THREAD

**Issue by hummuscience** [moderncv/moderncv#225]
**Describe the bug**
I am trying to create a credit table to describe what my contribution to each publication was using the following package: https://github.com/Pseudomanifold/latex-credits

This is based on these guidelines: https://credit.niso.org/

And could be useful to add to CVs to clarify what part of a sudy a person has contributed to. Here is an example:

![D-vQWpCUYAEhD2I](https://github.com/user-attachments/assets/b9e33f43-17b4-4d5e-9d11-8561784ded1e)

**Expected Behavior**
The expected behavior is to do the following in the preamble:

```
\usepackage{credits}
\credit{Doug}  {1,0.5,0,0,0.2,0.8,1,1,1,1,0.1,0,0.9,1}
\credit{Eve}    {0,1,0,1,0,1,0,1,0,1,0,1,0,1}
```

And then use the following in the text:

`\insertcredits`

And get a table similar to this:

![Screenshot 2024-12-11 at 19 08 41](https://github.com/user-attachments/assets/d7264860-f9d2-4968-913c-a4f07d003b0e)

**Actual Behavior**

What actually happens is the following:

![Screenshot 2024-12-11 at 19 10 22](https://github.com/user-attachments/assets/9204711a-202c-482c-92d6-4c2b1a791064)

There are no boxes visible. 

**Specifications**
  - moderncv version and source: freshly cloned today
  - LaTeX compiler (pdfLaTeX, XeLaTeX, etc.) and version: pdfLatex
  - Platform (operating system, LaTeX editor, etc.): overleaf/sharelatex

**Comment by LLdaniel** [comment]
Thanks for your report!
Can confirm your described behaviour. At a first glance I do not see why this happens. 
Maybe a clash of packages or package options can cause this. We should do some further investigations here!

**Comment by hummuscience** [comment]
Any idea where we could look? I could try some stuff out (but I am also clueless to latex details)

**Comment by LLdaniel** [comment]
I could not inspect it further yet, but in a first glance I just hardcoded a black fill in the `credit.sty` here (replaced `\cellfill` with `black`):
```latex
          \ifCREDITS@horizontal%
            \node[fill = \cellfill, draw = \CREDITS@grid, tile] at (\value{row}, -\authorIndex) {};
          \else%
            \node[fill = \cellfill, draw = \CREDITS@grid, tile] at (\authorIndex, -\value{row}) {};
          \fi%
```
This worked in principle. So I assumed that maybe determining the color within the loop leads to clashes with something in `moderncv`. I know that this is not a really deep analysis but I leave it here as you asked for it. 

At the moment I only can say that `moderncv` uses `tikz` for its *skillmatrix*, for *contemporary style* and for some icons that are not available through other packages.

**Comment by hummuscience** [comment]
When I hardcode black, I get all cells in black. So drawing is possible. The \cellfill seems to be defined as either white (default) or whatever was calculated here:

```
        \foreach \name/\column [count=\n] in \credits {%
          \foreach \entry [count = \m] in \column {%
            \pgfmathparse{\entry*100}
            \colorlet{fill}{\CREDITS@role!\pgfmathresult}
```
Which `\CREDITS@role `   defined as black in the beginnig of the file
I wonder why the calculation fails.

**Comment by hummuscience** [comment]
I just teasted the example which does not have the number in it, but just the names of the credits:

and that works, so its the calculation and fill that is failing

```
\documentclass{article}

\usepackage{xcolor}

\definecolor{cardinal} {RGB}{196, 30, 58}
\definecolor{lightgrey}{RGB}{150,150,150}

% You can configure the colour of the grid and the respective roles of
% individual authors.
\usepackage[role = cardinal, grid = lightgrey]{credits}

% The CRediT taxonomy
% - Conceptualization
% - Data curation
% - Formal analysis
% - Funding acquisition
% - Investigation
% - Methodology
% - Project administration
% - Resources
% - Software
% - Supervision
% - Validation
% - Visualization
% - Writing -- original draft
% - Writing -- review \& editing


% Enter the roles for each author as a comma-separated list with
% exact spelling. Order can be arbitrary.
\credit{Alice}{Conceptualization, Methodology, Supervision, Project administration, Funding acquisition, Writing -- review \& editing}
\credit{Bob}{Supervision, Conceptualization, Funding acquisition, Methodology, Resources, Writing -- review \& editing}
\credit{Charlie}{Software, conceptualization, Formal analysis, Writing -- original draft, Writing -- review \& editing, Validation, Visualization}
\credit{Doug}{Investigation, Formal analysis, Data curation, Writing -- review \& editing, Supervision}
\credit{Eve}{Data curation, Investigation, Software, Writing -- original draft}
\credit{Frank}{Investigation, Data curation, Software, Formal analysis}
\credit{Grace}{Data curation, Writing -- original draft, Visualization}
\credit{Heidi}{Investigation, Data curation}
\credit{Ivan}{Software, Visualization, Writing -- original draft}
\credit{Judy}{Investigation}
% Note the typo in Charlie's role "conceptualization", which is lowercase
% and thus not accepted, instead showing a warning.

\begin{document}
% Display the credits statement as a table
\insertcredits
\end{document}
```

**Comment by LLdaniel** [comment]
Ah, good, seems we came to the same conclusion!
Currently I use the following extended `template.tex`
```diff
└─▶ $ git diff template.tex
diff --git a/template.tex b/template.tex
index 5971baa..df4652c 100644
--- a/template.tex
+++ b/template.tex
@@ -8,9 +8,9 @@
 
 \documentclass[11pt,a4paper,sans]{moderncv}        % possible options include font size ('10pt', '11pt' and '12pt'), paper size ('a4paper', 'letterpaper', 'a5paper', 'legalpaper', 'executivepaper' and 'landscape') and font family ('sans' and 'roman')
 
-\moderncvcolor{cerulean}                           % color options 'black', 'blue' (default), 'burgundy', 'green', 'grey', 'orange', 'purple' and 'red'; for contemporary style use 'cerulean'
+\moderncvcolor{blue}                           % color options 'black', 'blue' (default), 'burgundy', 'green', 'grey', 'orange', 'purple' and 'red'; for contemporary style use 'cerulean'
 % moderncv themes
-\moderncvstyle[]{contemporary}                     % style options are 'casual' (default), 'classic', 'banking', 'oldstyle', 'fancy' and 'contemporary'
+\moderncvstyle[]{classic}                     % style options are 'casual' (default), 'classic', 'banking', 'oldstyle', 'fancy' and 'contemporary'
                                                    % the 'contemporary' style optionally takes the `qr` (default) or `noqr` options
 %\renewcommand{\familydefault}{\sfdefault}         % to set the default font; use '\sfdefault' for the default sans serif font, '\rmdefault' for the default roman one, or any tex font name
 %\nopagenumbers{}                                  % uncomment to suppress automatic page numbering for CVs longer than one page
@@ -51,6 +51,11 @@
 
 % document language
 \usepackage[english]{babel}  % FIXME: using spanish breaks moderncv
+\usepackage[role = red]{credits}
+%\credit{Alice}{Conceptualization, Methodology, Supervision, Project administration, Funding acquisition, Writing -- review \& editing}
+%\credit{Bob}{Supervision, Conceptualization, Funding acquisition, Methodology, Resources, Writing -- review \& editing}
+\credit{Doug}{1,0.5,0,0,0.2,0.8,1,1,1,1,0.1,0,0.9,1}
+\credit{Eve}{0,1,0,1,0,1,0,1,0,1,0,1,0,1}
 
 % personal data
 \name{John}{Doe}
@@ -131,7 +136,7 @@
 
 \section{Education}  % for 'contemporary' style use optional argument for displaying an icon, e.g. \section[\faGraduationCap]{Education}
 \cventry{year--year}{Degree}{Institution}{City}{\textit{Grade}}{Description}  % arguments 3 to 6 can be left empty
-\cventry{year--year}{Degree}{Institution}{City}{\textit{Grade}}{Description}
+\cventry{year--year}{Degree}{Institution}{City}{\textit{Grade}}{\insertcredits}
 
 \section{Master thesis}
 \cvitem{title}{\emph{Title}}
```

I am not sure if it is right that I am in the following loop when compiling with `lualatex`:
https://github.com/Pseudomanifold/latex-credits/blob/04c5805a0ef83728e8f0a749bb77d610aafbc615/credits.sty#L196-L203
I added `typeout` statements like this:
```diff
└─▶ $ diff -u credits.sty credits.sty_orig 
--- credits.sty	2024-12-19 13:16:08.203126990 +0100
+++ credits.sty_orig	2024-12-19 12:33:50.612233202 +0100
@@ -196,11 +196,9 @@
         % Now for each author, check if they contributed to this concept
         \foreach \name/\column [count=\authorIndex] in \credits {%
           \foreach \entry in \column {%
-          \typeout{credits entry: \entry, credits concept: \concept}
             % Default color
             \def\cellfill{white}%
             \ifthenelse{\equal{\entry}{\concept}}{%
-              \typeout{ === MATCH FOUND ===}
               \def\cellfill{\CREDITS@role}%
               \breakforeach
             }{}%
@@ -366,4 +364,4 @@
     {}%
     \fi
   }%
-}
+}
\ No newline at end of file
```

I am not able to get a match especially I am wondering why there is the comparision between the *number entry* and the *concept string*. See my compile output (or log):
```
credits entry: 1, credits concept: Conceptualization
credits entry: 0.5, credits concept: Conceptualization
credits entry: 0, credits concept: Conceptualization
credits entry: 0, credits concept: Conceptualization
credits entry: 0.2, credits concept: Conceptualization
credits entry: 0.8, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 0.1, credits concept: Conceptualization
credits entry: 0, credits concept: Conceptualization
credits entry: 0.9, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 0, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 0, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 0, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 0, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 0, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 0, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 0, credits concept: Conceptualization
credits entry: 1, credits concept: Conceptualization
credits entry: 1, credits concept: Data curation
credits entry: 0.5, credits concept: Data curation

[...]
```

I am not sure if it is intended that I run through this *foreach*, because this seems not suitable when using *floats*.
On the other hand when I use the modified `credits.sty` with the official granular example my `typeouts` are not shown, because then I am running through the following *foreach*:
https://github.com/Pseudomanifold/latex-credits/blob/04c5805a0ef83728e8f0a749bb77d610aafbc615/credits.sty#L264-L279


That gave me the hint, to use `\cventry{year--year}{Degree}{Institution}{City}{\textit{Grade}}{\insertcreditsgranular}` instead of `\cventry{year--year}{Degree}{Institution}{City}{\textit{Grade}}{\insertcredits}`. 
This created the following output:
![image](https://github.com/user-attachments/assets/735e98ce-02be-44f5-86ca-ba23ab335cdb)

**tldr; I suspect `insertcreditsgranular` is needed instead of `insertcredits`. Could you please verify that? Seems that there is no problem with `moderncv`, but with calling the correct command for *granular credits*.**

**Comment by hummuscience** [comment]
Yes! That solves it. Thank you!
